### PR TITLE
[Fixes #7507] Owners API V2 Returns users with no resources

### LINKED ISSUE #7507: Owners API V2 Returns users with no resources
## Expected Behavior
The owners API should return only users with at least 1 resource

## Actual Behavior
The owners API returns all users including those who don't own any resource

## Steps to Reproduce the Problem

  1.https://master.demo.geonode.org/api/v2/owners
## Specifications

  - GeoNode version: master
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker):
  - Platform:
  - Additional details:

### PR DESCRIPTION
**Pull request by marthamareal**
<Include a few sentences describing the overall goals for this Pull Request>
References #7507

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/base/api/views.py [hunk 1]
```diff
@@ -198,10 +198,26 @@ class OwnerViewSet(WithDynamicViewSetMixin, ListModelMixin, GenericViewSet):
...
+        Filter users with atleast a resource
+        """
+        queryset = get_user_model().objects.all().exclude(pk=-1)
+        filter_options = {}
+        if self.request.query_params:
+            filter_options = {
+                'type_filter': self.request.query_params.get('type'),
+                'title_filter': self.request.query_params.get('title__icontains')
+                }
+        resources_ids = get_resources_with_perms(
```
Comment (giohappy) [inline geonode/base/api/views.py:215]: shouldn't this variabile be named owner_ids @marthamareal?

**Review by giohappy — CHANGES_REQUESTED** [review]

File: geonode/base/api/views.py (hunk 1 above)
Comment (marthamareal) [reply geonode/base/api/views.py]: correct.i have fixed it.

File: geonode/base/api/views.py [hunk 2]
```diff
@@ -198,10 +198,26 @@ class OwnerViewSet(WithDynamicViewSetMixin, ListModelMixin, GenericViewSet):
...
+        Filter users with atleast a resource
+        """
+        queryset = get_user_model().objects.all().exclude(pk=-1)
+        filter_options = {}
+        if self.request.query_params:
+            filter_options = {
+                'type_filter': self.request.query_params.get('type'),
+                'title_filter': self.request.query_params.get('title__icontains')
+                }
+        owner_ids = get_resources_with_perms(
```
Comment (giohappy) [inline geonode/base/api/views.py:215]: why don't you use a Subquery? Your solution will issue two DB queries, doesn't it?
https://docs.djangoproject.com/en/2.2/ref/models/expressions/#subquery-expressions

**Review by giohappy — CHANGES_REQUESTED** [review]

File: geonode/base/api/views.py (hunk 2 above)
Comment (marthamareal) [reply geonode/base/api/views.py]: updated

**Review by afabiani — APPROVED** [review]

**Pull request merged**